test/passes/flatten_all-features.txt

```diff
@@ -2443,3 +2443,20 @@
   (unreachable)
  )
 )
+(module
+ (type $none_=>_none (func))
+ (type $none_=>_funcref (func (result funcref)))
+ (func $0 (result funcref)
+  (local $0 (ref null $none_=>_none))
+  (local.set $0
+   (ref.as_non_null
```

Comment:
I wonder why was this `ref.as_non_null` generated? I thought they are generated only when we do `local.get`.

Reply:
I think this `ref.as_non_null` was already present in the input. It's only the later one that is added.

Reply:
Oh right, sorry...